### ISSUE TITLE
Add an optional transformer function to ValueNotifierAdapter

### ISSUE THREAD

**Issue by gskinner** [gskinner/flutter_animate#27]
It may occasionally be useful to be able to specify a function to transform the notifier value before it is assigned to the controller.

**Comment by himanshugarg08** [comment]
Yeah, makes sense. But that transformation may not always be in the range `[0-1]`. So the result should be clamped between 0 and 1?
